### ISSUE TITLE
How can I help?

### ISSUE THREAD

**Issue by adamcharnock** [jondurbin/airoboros#25]
I was wondering if you are looking for any help? I very much like the concept of this project, and have read a good portion of the code and it makes sense to me at a high-level.

My motivation is that I want to work to contribute to a project that is making an impact, and I also want to expand my experience in the generative AI field.

Quick summary of my pros/cons (being as direct as possible):

* **Pro**: I'm very experienced in Python, including asyncio ([relevant example](https://github.com/adamcharnock/lightbus))
* **Pro**: I'm a good communicator and happy writing docs ([same project](https://lightbus.org/))
* **Con**: I'm rather new to the Generative AI field. I grasp many of the high level concepts but will struggle reading the academic research
* **Con**: My availability varies with my freelance work. Right now I have 1-2 days a week available, but that could change.

Ways in which I could help:

1. Develop new features
2. Improve any code you think would benefit from it
3. Write docs

Let me know if any of this would be of use to you. If so, we can figure out what to do next. Of course, no worries if not.

**Comment by jondurbin** [comment]
Thanks Adam, very kind to offer.  The project is in desperate need of documentation, contributing guidelines, etc., that I just haven't had time to get to yet, so that could be very helpful.

The list of instructors could also be extended quite a bit so have a much broader range of options.  Off the top of my head, I would think a text editing option would be good.  For example, a way to create a poorly written piece of text with misspelling, incorrect grammar/casing/etc. then rewrite it to be correct.  Another option would be to take a short, less detailed piece of text and extend it to have longer, more detailed and intelligent writing, etc.

Basically any ideas you might have for specific training data that could enhance the model could be useful.

**Comment by adamcharnock** [comment]
Amazing, thank you for such a quick reply. That all sounds very good and sensible to me. It is bed time for me here, but I'll pick this up again sometime tomorrow.

**Comment by adamcharnock** [comment]
Ok, so running with what you said, potential tasks may look something like:

# Documentation

  - [ ] Create an overall documentation framework/site, followed by...
  - [ ] Contribution guidelines
  - [ ] Getting started docs (I would write these while also getting myself spun-up on the project)
  - [ ] Reference docs
    - [ ] Docs for each trained model (current status, motivation, leanings, hardware used, training time)
    - [ ] Docs for each instructor
  - [ ] Explanation / howto docs

## Documentation framework

I have previously used [David Laing's documentation framework](https://documentation.divio.com) and found it to be really very effective. This is because:

* It provides clear structure for the reader, and for readers with different needs
* It also provides clear structure for the writer. It answers the "I need to document this, but where should it go?" question.

My plan would be to create the structure, write as much as I can, and occasionally call on you for draft content (either written, or I could interview you and write it myself)

# Development

* Instructors
* Other infrastructure code (e.g. API, etc) – I'm actually pretty keen on this if something comes up.

# Next steps

If this sounds good to you then I will create a new issue to track documentation creation. In this issue I'll propose a structure and a documentation system.

**Comment by jondurbin** [comment]
Yeah, looks great, thank you!  Even a structural outline with your understanding or simple stubs would go a long way in helping!

**Comment by adamcharnock** [comment]
Tracking progress in #28

**Comment by jackdiy** [comment]
Seeing your exchange is very inspiring. 
Personally, I don't know much about technology, but I must express my gratitude here.
Thank you very much for your hard work, and I think the fans here would be very interested in getting a complete detailed documentation and demonstration of LMOE.
